### PR TITLE
Feat/iterator pattern

### PR DESCRIPTION
**Pull request by daiarn**
Iterator pattern. Many changes on server side

### CODE REVIEW

File: UNO_Server/Models/SendData/GameSpectatorState.cs [hunk 1]
```diff
@@ -20,10 +20,10 @@ public class GameSpectatorState
 
 		public GameSpectatorState(Game game)
 		{
-			zeroCounter = game.observers[0].Counter;
-			wildCounter = game.observers[1].Counter;
+            zeroCounter = game.gameWatcher.observers[0].Counter;
+            wildCounter = game.gameWatcher.observers[1].Counter;
 
-			phase = game.phase;
+            phase = game.phase;
```
Comment (daiarn) [inline UNO_Server/Models/SendData/GameSpectatorState.cs:26]: Spacing looks weird but on my pc it is normal

File: UNO_Server/Models/Game.cs [hunk 2]
```diff
@@ -214,10 +210,12 @@ public bool CanCardBePlayed(Card playerCard)
...
-			foreach (var item in observers)
-			{
-				item.Notify(card);
-			}
+            var iterator = gameWatcher.GetIterator();
+            while (iterator.HasNext)
+            {
+                Observer observer = (Observer)iterator.Next;
+                observer.Notify(card);
+            }
```
Comment (daiarn) [inline UNO_Server/Models/Game.cs:221]: Weird spacing again

Comment (ghost) [reply UNO_Server/Models/Game.cs:221]: put all cardsCounter.AddCard methods in here and not all over the place

**Review by ghost — CHANGES_REQUESTED** [review]
See comment on Game class

File: UNO_Server/Models/Game.cs (hunk 2 above)
Comment (daiarn) [reply UNO_Server/Models/Game.cs:221]: hot fix incoming

**Comment by ghost** [comment]
Iterator works, not sure about the other stuff

**Pull request merged**